src/Artifacts.jl

```diff
@@ -667,8 +668,12 @@ end
 
 Download/install an artifact into the artifact store.  Returns `true` on success.
 """
-function download_artifact(tree_hash::SHA1, tarball_url::String, tarball_hash::String;
-                           verbose::Bool = false)
+function download_artifact(
+    tree_hash::SHA1,
+    tarball_url::String,
+    tarball_hash::Union{String, Nothing} = nothing;
```

Comment:
Because we have no way of getting a tarball hash for Pkg protocol downloads.

Reply:
A TODO item here is to, once we have better integration in our downloading stack (e.g. using `HTTP.jl`) to write a `secure_download()` function that either (a) ensures that you are downloading over HTTPS, or (b) has a hash to verify the content, and refuses to download if either of those things are not true.

Reply:
Pkg protocol will officially not support plain HTTP, although currently we're not enforcing that and I'm testing over HTTP locally because HTTPS is annoying to setup. But maybe one of these should be true:
1. There's a download hash to check that you're getting what you expect
2. You're talking to localhost
3. You're using HTTPS

Reply:
Yes, that seems reasonable to me, and much easier to enforce in pure Julia than using something like `curl`.